std/utf.d

```diff
@@ -2426,11 +2426,6 @@ pure
 {
 
 char[] toUTF8(return out char[4] buf, dchar c) nothrow @nogc @safe
-in
-{
-    assert(isValidDchar(c));
```

Comment:
How does this violate memory safety, and how does removing this assert help fix the problem?